deepspeed/pt/replace_module.py

```diff
@@ -0,0 +1,199 @@
+import copy
+import torch
+import deepspeed
+
+
+def replace_transformer_layer(orig_layer_impl,
```

Comment:
Same comment as above. When `huggingface=False`, which transformer implementation are we assuming?

This seems like the place where @tjruwase 's idea could be implemented. Totally brainstormed and untested:

```python
from abc import ABC
class TransformerReplacementPolicy(ABC):
    @classmethod
    def replace(other):
        pass
        # the main replacement body here, but use methods like self.query(other)

    @abstractmethod    
    def query(self, source):
        """"Should return a tuple of the weight and bias for the query. """
        pass

class HFTransformerReplacement(TransformerReplacementPolicy):
    def query(self, source):
        qw = source.self.query.weight
        qb = source.self.query.bias
        return (qw, qb)
```